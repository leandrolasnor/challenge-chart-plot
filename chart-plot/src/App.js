import React, { Component } from 'react'
import './common/adminLTE/dependences'
import Header from "./common/adminLTE/header"
import Main from "./common/adminLTE/main"
import Footer from "./common/adminLTE/footer"
import Messages from "./common/msg/messages"
import Dashboard from './dashboard/dashboard'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {dispatch_data_charts} from './editor/actions'

class App extends Component {

    generate_chart = () => {
        const {editor} = this.props
        this.props.dispatch_data_charts(editor.inputs)
    }

    render() {
        return (
            <div className="wrapper">
                <Header />
                <Main>
                    <div className="content-wrapper">
                        <Dashboard />
                    </div>
                    <Footer btnHandleClick={this.generate_chart} btnText="GENERATE CHART" />
                    <Messages />
                </Main>
            </div>
        );
    }
}
const mapDispatchToProps = dispatch => bindActionCreators({dispatch_data_charts}, dispatch)
const mapStateToProps = state => ({editor: state.editor})
export default connect(mapStateToProps, mapDispatchToProps)(App)
