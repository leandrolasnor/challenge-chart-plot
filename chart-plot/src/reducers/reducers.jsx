import { combineReducers } from 'redux'
import DashboardReducer from '../dashboard/reducer'
import CodeReducer from '../code/reducer'
import { reducer as toastrReducer } from 'react-redux-toastr'

const rootReducer = combineReducers({
    dashboard: DashboardReducer,
    editor: CodeReducer,
    toastr: toastrReducer
})

export default rootReducer