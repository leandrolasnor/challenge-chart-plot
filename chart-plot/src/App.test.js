import React from "react";
import ReactDOM from "react-dom";
import { applyMiddleware, createStore } from "redux";
import { Provider } from "react-redux";
import promise from "redux-promise";
import multi from "redux-multi";
import thunk from "redux-thunk";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import reducers from "./reducers/reducers";
import App from "./App";
import Editor from "./editor/editor";
import Dashboard from "./dashboard/dashboard";

const store = applyMiddleware(multi, thunk, promise)(createStore)(reducers);

it("<App />", () => {
  expect(toJson(shallow(<Provider store={store}>{App}</Provider>))).toMatchSnapshot();
});

it("<Editor />", () => {
  expect(toJson(shallow(<Provider store={store}>{Editor}</Provider>))).toMatchSnapshot();
});

it("<Dashboard />", () => {
  expect(toJson(shallow(<Provider store={store}>{Dashboard}</Provider>))).toMatchSnapshot();
});
