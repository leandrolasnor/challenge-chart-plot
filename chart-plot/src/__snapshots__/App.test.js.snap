// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "chart": Object {
          "data": Object {
            "datasets": Array [],
            "labels": Array [],
          },
          "executing": false,
          "group": Array [],
          "haveEventsPlotted": false,
          "options": Object {
            "scales": Object {
              "xAxes": Array [
                Object {
                  "ticks": Object {
                    "callback": [Function],
                  },
                },
                Object {
                  "display": true,
                  "position": "right",
                },
              ],
            },
            "time": Object {
              "max": 9999999999999,
              "min": 0,
            },
          },
          "select": Array [],
          "span": Object {
            "begin": 0,
            "end": 9999999999999,
          },
        },
        "editor": Object {
          "blocked": false,
          "executing": false,
          "inputs": "{\\"type\\": \\"start\\", \\"timestamp\\": \\"1519862400000\\", \\"select\\": [\\"min_response_time\\", \\"max_response_time\\"], \\"group\\":[\\"os\\", \\"browser\\"]}
{\\"type\\": \\"span\\", \\"timestamp\\": \\"1519862400000\\", \\"begin\\": \\"1519862400000\\", \\"end\\": \\"1519862460000\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.3\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"1.2\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.3\\", \\"max_response_time\\":\\"1.2\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.0\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"0.9\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.0\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"1.1\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.3\\", \\"max_response_time\\":\\"1.4\\"}
{\\"type\\": \\"stop\\", \\"timestamp\\": \\"1519862460000\\"}",
        },
        "toastr": Object {
          "confirm": null,
          "toastrs": Array [],
        },
      },
    }
  }
>
  <Component />
</ContextProvider>
`;

exports[`<Dashboard /> 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "chart": Object {
          "data": Object {
            "datasets": Array [],
            "labels": Array [],
          },
          "executing": false,
          "group": Array [],
          "haveEventsPlotted": false,
          "options": Object {
            "scales": Object {
              "xAxes": Array [
                Object {
                  "ticks": Object {
                    "callback": [Function],
                  },
                },
                Object {
                  "display": true,
                  "position": "right",
                },
              ],
            },
            "time": Object {
              "max": 9999999999999,
              "min": 0,
            },
          },
          "select": Array [],
          "span": Object {
            "begin": 0,
            "end": 9999999999999,
          },
        },
        "editor": Object {
          "blocked": false,
          "executing": false,
          "inputs": "{\\"type\\": \\"start\\", \\"timestamp\\": \\"1519862400000\\", \\"select\\": [\\"min_response_time\\", \\"max_response_time\\"], \\"group\\":[\\"os\\", \\"browser\\"]}
{\\"type\\": \\"span\\", \\"timestamp\\": \\"1519862400000\\", \\"begin\\": \\"1519862400000\\", \\"end\\": \\"1519862460000\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.3\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"1.2\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.3\\", \\"max_response_time\\":\\"1.2\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.0\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"0.9\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.0\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"1.1\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.3\\", \\"max_response_time\\":\\"1.4\\"}
{\\"type\\": \\"stop\\", \\"timestamp\\": \\"1519862460000\\"}",
        },
        "toastr": Object {
          "confirm": null,
          "toastrs": Array [],
        },
      },
    }
  }
>
  <Component />
</ContextProvider>
`;

exports[`<Editor /> 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "chart": Object {
          "data": Object {
            "datasets": Array [],
            "labels": Array [],
          },
          "executing": false,
          "group": Array [],
          "haveEventsPlotted": false,
          "options": Object {
            "scales": Object {
              "xAxes": Array [
                Object {
                  "ticks": Object {
                    "callback": [Function],
                  },
                },
                Object {
                  "display": true,
                  "position": "right",
                },
              ],
            },
            "time": Object {
              "max": 9999999999999,
              "min": 0,
            },
          },
          "select": Array [],
          "span": Object {
            "begin": 0,
            "end": 9999999999999,
          },
        },
        "editor": Object {
          "blocked": false,
          "executing": false,
          "inputs": "{\\"type\\": \\"start\\", \\"timestamp\\": \\"1519862400000\\", \\"select\\": [\\"min_response_time\\", \\"max_response_time\\"], \\"group\\":[\\"os\\", \\"browser\\"]}
{\\"type\\": \\"span\\", \\"timestamp\\": \\"1519862400000\\", \\"begin\\": \\"1519862400000\\", \\"end\\": \\"1519862460000\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.3\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"1.2\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.3\\", \\"max_response_time\\":\\"1.2\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862400000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.0\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"0.9\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"chrome\\", \\"min_response_time\\":\\"0.1\\", \\"max_response_time\\":\\"1.0\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"mac\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.2\\", \\"max_response_time\\":\\"1.1\\"}
{\\"type\\": \\"data\\", \\"timestamp\\": \\"1519862460000\\", \\"os\\": \\"linux\\", \\"browser\\": \\"firefox\\", \\"min_response_time\\":\\"0.3\\", \\"max_response_time\\":\\"1.4\\"}
{\\"type\\": \\"stop\\", \\"timestamp\\": \\"1519862460000\\"}",
        },
        "toastr": Object {
          "confirm": null,
          "toastrs": Array [],
        },
      },
    }
  }
>
  <Component />
</ContextProvider>
`;
